src/openrct2/world/footpath.c

```diff
@@ -441,19 +442,38 @@ money32 footpath_remove_real(sint32 x, sint32 y, sint32 z, sint32 flags)
             network_set_player_last_action_coord(network_get_player_index(game_command_playerid), coord);
         }
 
-        footpath_queue_chain_reset();
-        remove_banners_at_element(x, y, mapElement);
-        footpath_remove_edges_at(x, y, mapElement);
-        map_invalidate_tile_full(x, y);
-        map_element_remove(mapElement);
-        footpath_update_queue_chains();
+        // If the ghost flag is present we have to make sure to only delete ghost footpaths as they may be
+        // at the same origin.
+        if ((flags & GAME_COMMAND_FLAG_GHOST) && map_element_is_ghost(mapElement) == false)
+        {
+            while (!map_element_is_last_for_tile(mapElement++))
+            {
+                if (mapElement->type != MAP_ELEMENT_TYPE_PATH) continue;
```

Comment:
Could you change these one-liners to 'normal' if statements with brackets? It's better for legibility.